dev/generate_test_trajectory.ts

```diff
@@ -1,144 +1,450 @@
+    resultTrajectory.timestamps.push(null)
+  }
+  return resultTrajectory
+}
+
+function addTimestampsForTrajectory(
+  first: Date,
+  last: Date,
+  trajectory: Trajectory
+): Trajectory {
+  const duration = last.getTime() - first.getTime()
+  const numberTimestamps = trajectory.coordinates.length
+  const stepLength = duration / numberTimestamps
+  const resultTrajectory = trajectory.getCopy()
+  for (let i = 0; i < numberTimestamps; i++) {
+    resultTrajectory.timestamps[i] = new Date(first.getTime() + i * stepLength)
   }
-  return trajectory
+  return resultTrajectory
+}
+
+/**
+ * Interpolates and adds additional coordinates in between given set of coordinates.
+ * @param trajectory which holds coordinates
+ * @param numberOfInsertsPerCoordinate number of interpolated points between each pair of subsequent coordinates
+ * @returns interpolated trajectory
+ */
+function interpolateCoordinates(
+  trajectory: Trajectory,
+  numberOfInsertsPerCoordinate: number = 1
+) {
+  if (numberOfInsertsPerCoordinate < 1) {
+    return trajectory
+  }
+  const resultTrajectory = trajectory.getCopy()
+  const numberCoordinates = trajectory.coordinates.length
+  for (let index = 0, indexShift = 0; index < numberCoordinates - 1; index++) {
+    const [firstLat, firstLng] = trajectory.coordinates[index]
+    const [secondLat, secondLng] = trajectory.coordinates[index + 1]
+    const latDiff = firstLat - secondLat
+    const lngDiff = firstLng - secondLng
+    for (
+      let insertIndex = 1;
+      insertIndex <= numberOfInsertsPerCoordinate;
+      insertIndex++, indexShift++
+    ) {
+      const latSeed =
+        firstLat + latDiff * (insertIndex / numberOfInsertsPerCoordinate)
+      const lngSeed =
+        firstLng + lngDiff * (insertIndex / numberOfInsertsPerCoordinate)
+      const insertCoordinate = randomGeo(latSeed, lngSeed, 0.25, 0.5)
+      resultTrajectory.coordinates.splice(index + indexShift + 1, 0, [
+        insertCoordinate.latitude,
+        insertCoordinate.longitude,
+      ])
+      resultTrajectory.timestamps.splice(index + indexShift + 1, 0, null)
+    }
+  }
+  return resultTrajectory
 }
 
 function randomGeo(
   latitude: number,
   longitude: number,
-  radiusInMeters: number
+  minRadiusInMeters: number,
+  maxRadiusInMeters: number
 ) {
-  var y0 = latitude
-  var x0 = longitude
-  var rd = radiusInMeters / 111300
+  const y0 = latitude
+  const x0 = longitude
+  const randRadius =
+    (minRadiusInMeters +
+      Math.random() * (maxRadiusInMeters - minRadiusInMeters)) *
+    randomSign()
+  const rd = randRadius / 111300
 
-  var u = Math.random()
-  var v = Math.random()
+  const u = Math.random()
+  const v = Math.random()
 
-  var w = rd * Math.sqrt(u)
-  var t = 2 * Math.PI * v
-  var x = w * Math.cos(t)
-  var y = w * Math.sin(t)
+  const w = rd * Math.sqrt(u)
+  const t = 2 * Math.PI * v
+  const x = w * Math.cos(t)
+  const y = w * Math.sin(t)
+  const signedX = x * randomSign()
+  const signedY = y * randomSign()
```

Comment:
I think we don't need a random sign here, since `Math.sin` and `Math.cos` can also produce negative values depending on the input. Or did you encounter a bias towards positive directions during testing?

Reply:
You're right. Thought it would be a good idea to randomize the sign, without paying to much attention to the prior code which already accounts for that.